systemd/systemd.go

```diff
@@ -1064,9 +1112,21 @@ LazyUnmount=yes
 WantedBy=multi-user.target
 `
 
-func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
-	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
-	mu := MountUnitPath(where)
+const (
+	snappyCreatorModule = "X-SnapdOrigin"
+)
+
+func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+	snapDesc := u.SnapName
```

Comment:
should this have a block initializing the options or can this function not function at all with entirely empty options? If it's the latter case, we should return an error if u is nil I think, otherwise this will just panic

Reply:
I think that a panic here is more appropriate, because calling this method with a `nil` as second parameter is a programming error. But I've added an `if`to make it more explicit.

Reply:
thanks, looks good to me